arduino/examples/ESP32IOBoard/ESP32IOBoard.ino

```diff
@@ -269,16 +282,35 @@ void setup()
         }
     }
 
+    // initialize all declared GPIO pins
+    GpioInit::hw_init();
+
+#if defined(FACTORY_RESET_GPIO_PIN)
+    // Check the factory reset pin which should normally read HIGH (set), if it
+    // reads LOW (clr) delete the cdi.xml and openlcb_config
+    if (FACTORY_RESET_pin::instance()->is_clr())
+    {
+        printf("!!!! WARNING WARNING WARNING WARNING WARNING !!!!\n");
+        printf("The factory reset GPIO pin %d has been triggered.\n",
+               FACTORY_RESET_GPIO_PIN);
+        for (uint8_t sec = 10; sec > 0; sec--)
```

Comment:
There does not seem to be any way to exit from this loop. Shouldn't we check that the pin is still held down?

Reply:
Are you suggesting that we should verify on each loop that they still have the pin held LOW and if it is not still LOW at the end we do not reset?